ocs/ocp.py

```diff
@@ -0,0 +1,156 @@
+"""
+General OCP object
+"""
+import os
+import logging
+import yaml
+
+from munch import munchify
+from utility.utils import TimeoutSampler
+from utility.utils import run_cmd
+
+log = logging.getLogger(__name__)
+
+
+class OCP(object):
+    """
+    A basic OCP object to run basic 'oc' commands
+    """
+
+    def __init__(self, api_version='v1', kind='Service', namespace=None):
+        """
+        Initializer function
+
+        Args:
+            api_version (str): TBD
+            kind (str): TBD
+            namespace (str): The name of the namespace to use
+        """
+        self._api_version = api_version
+        self._kind = kind
+        self._namespace = namespace
+
+    @property
+    def api_version(self):
+        return self._api_version
+
+    @property
+    def kind(self):
+        return self._kind
+
+    @property
+    def namespace(self):
+        return self._namespace
+
+    def exec_oc_cmd(self, command):
+        """
+        Executing 'oc' command
+
+        Args:
+            command (str): The command to execute (e.g. create -f file.yaml)
+                without the initial 'oc' at the beginning
+
+        Returns:
+            Munch Obj: this object represents a returned yaml file
+        """
+        oc_cmd = "oc "
+        kubeconfig = os.getenv('KUBECONFIG')
+        if self.namespace:
+            oc_cmd += f"-n {self.namespace} "
+
+        if kubeconfig:
+            oc_cmd += f"--kubeconfig {kubeconfig} "
+
+        oc_cmd += command
+        out = run_cmd(cmd=oc_cmd)
+        return munchify(yaml.safe_load(out))
+
+    def get(self, resource_name='', out_yaml_format=True):
```

Comment:
what about to user to pass additional extra parameters to this oc get commonad (not only in `oc get` but others as well). 

I suggest like additional parameter `parameters` which can be list of the optional parameters as we probably don't want to enable all functionality here. But to give the user to option use more that is defined here? (Of course on user responsibility, so if they will add parameter --output json they will break the behavior here)

so if user will add parameters=['--request-timeout', 60] for example they can change the behavior of the command regarding the [documentation](https://www.mankier.com/1/oc-get#--request-timeout)

Reply:
I agree but because this is an initial PR I will leave the advanced implementation to a different PR.
We can always extend it